pkg/p2p/libp2p/libp2p.go

```diff
@@ -326,6 +309,10 @@ func (s *Service) Peers() []p2p.Peer {
 	return s.peers.peers()
 }
 
+func (s *Service) SetPeerAddedHandler(h func(ctx context.Context, addr swarm.Address) error) {
```

Comment:
no need to declare `ctx` and `addr`, just the types would do. nicer would be to actually have the signature as a type, and just use that around since it is used in several places